src/pybind/mgr/cephadm/module.py

```diff
@@ -1398,6 +1398,11 @@ def _schedulable_hosts(self) -> List[HostSpec]:
         ]
 
     def _check_valid_addr(self, host: str, addr: str) -> str:
+        #make sure mgr is not resolving own ip
+        if addr in self.get_mgr_id():
+            raise OrchestratorError("can not resolve ip of host running active mgr please supply addr")
```

Comment:
NIT, I think the message could be a bit clearer
```suggestion
            raise OrchestratorError("Can not automatically resolve ip address of host where active mgr is running. Please explicitly provide the address.")
```